### PR TITLE
Update foreman-tasks to 0.8.6 (RPM)

### PR DESCRIPTION
**Pull request by iNecas**
Depends on https://github.com/theforeman/foreman-packaging/pull/1422

### CODE REVIEW

File: rubygem-foreman-tasks/rubygem-foreman-tasks.spec [hunk 1]
```diff
@@ -26,7 +26,7 @@ Source0: http://rubygems.org/downloads/%{gem_name}-%{version}.gem
 Requires: foreman >= 1.9.0
 
 Requires: %{?scl_prefix}rubygem(foreman-tasks-core)
-Requires: %{?scl_prefix}rubygem(dynflow) >= 0.8.13
```
Comment (domcleal) [inline rubygem-foreman-tasks/rubygem-foreman-tasks.spec]: missing identical change to BRs

File: rubygem-foreman-tasks/rubygem-foreman-tasks.spec [hunk 2]
```diff
@@ -165,13 +168,14 @@ exit 0
 %config %{foreman_pluginconf_dir}/%{gem_name}.yaml
 %{foreman_apipie_cache_foreman}
 %{foreman_apipie_cache_plugin}
+%{foreman_dir}/script/foreman-debug.d/60-dynflow_debug
 %doc %{gem_instdir}/LICENSE
 %{bin_dir}/%{service_name}
+config %{sysconfig_dir}/%{service_name}
```
Comment (domcleal) [inline rubygem-foreman-tasks/rubygem-foreman-tasks.spec]: missing % for %config

File: rubygem-foreman-tasks/rubygem-foreman-tasks.spec [hunk 3]
```diff
@@ -105,15 +105,18 @@ mv %{buildroot}/%{gem_instdir}/config/%{gem_name}.yaml.example \
...
 install -Dp -m0755 %{buildroot}%{gem_instdir}/%{confdir}/%{service_name}.init %{buildroot}%{_root_initddir}/%{service_name}
 %else
-install -Dp -m0644 %{buildroot}%{gem_instdir}/%{confdir}/%{service_name}.sysconfig %{buildroot}%{sysconfig_dir}/%{service_name}
 install -Dp -m0644 %{buildroot}%{gem_instdir}/%{confdir}/%{service_name}.service %{buildroot}%{_unitdir}/%{service_name}.service
 %endif
 mkdir -p %{buildroot}%{bin_dir}
 ln -sv %{gem_instdir}/bin/%{service_name} %{buildroot}%{bin_dir}/%{service_name}
 
+#link dynflow-debug.sh to be called from foreman-debug
+ln -s %{gem_instdir}/extra/dynflow-debug.sh %{foreman_dir}/script/foreman-debug.d/60-dynflow_debug
```
Comment (domcleal) [inline rubygem-foreman-tasks/rubygem-foreman-tasks.spec]: Though it looks correctly set up for now, I'd suggest either using an explicit chmod +x on the target file, or using `install` (or %files) to install the script into %{foreman_dir} with +x permissions to ensure it remains correct.

**Comment by iNecas** [comment]
Updated

**Comment by iNecas** [comment]
[test]

File: rubygem-foreman-tasks/rubygem-foreman-tasks.spec [hunk 4]
```diff
@@ -105,15 +105,19 @@ mv %{buildroot}/%{gem_instdir}/config/%{gem_name}.yaml.example \
...
+chmod +x %{foreman_dir}/script/foreman-debug.d/60-dynflow_debug
```
Comment (domcleal) [inline rubygem-foreman-tasks/rubygem-foreman-tasks.spec]: This file is a symlink, `chmod +x` is superfluous - it should be the _target_ file, extra/dynflow-debug.sh.

File: rubygem-foreman-tasks/rubygem-foreman-tasks.spec [hunk 5]
```diff
@@ -105,15 +105,19 @@ mv %{buildroot}/%{gem_instdir}/config/%{gem_name}.yaml.example \
...
 install -Dp -m0755 %{buildroot}%{gem_instdir}/%{confdir}/%{service_name}.init %{buildroot}%{_root_initddir}/%{service_name}
 %else
-install -Dp -m0644 %{buildroot}%{gem_instdir}/%{confdir}/%{service_name}.sysconfig %{buildroot}%{sysconfig_dir}/%{service_name}
 install -Dp -m0644 %{buildroot}%{gem_instdir}/%{confdir}/%{service_name}.service %{buildroot}%{_unitdir}/%{service_name}.service
 %endif
 mkdir -p %{buildroot}%{bin_dir}
 ln -sv %{gem_instdir}/bin/%{service_name} %{buildroot}%{bin_dir}/%{service_name}
 
+#link dynflow-debug.sh to be called from foreman-debug
+chmod +x %{gem_instdir}/extra/dynflow-debug.sh
```
Comment (domcleal) [inline rubygem-foreman-tasks/rubygem-foreman-tasks.spec]: `%{buildroot}%{gem_instdir}` should fix the current build error.

Comment (iNecas) [inline rubygem-foreman-tasks/rubygem-foreman-tasks.spec]: updated

File: rubygem-foreman-tasks/rubygem-foreman-tasks.spec [hunk 6]
```diff
@@ -105,15 +105,19 @@ mv %{buildroot}/%{gem_instdir}/config/%{gem_name}.yaml.example \
...
 %else
-install -Dp -m0644 %{buildroot}%{gem_instdir}/%{confdir}/%{service_name}.sysconfig %{buildroot}%{sysconfig_dir}/%{service_name}
 install -Dp -m0644 %{buildroot}%{gem_instdir}/%{confdir}/%{service_name}.service %{buildroot}%{_unitdir}/%{service_name}.service
 %endif
 mkdir -p %{buildroot}%{bin_dir}
 ln -sv %{gem_instdir}/bin/%{service_name} %{buildroot}%{bin_dir}/%{service_name}
 
+#link dynflow-debug.sh to be called from foreman-debug
+chmod +x %{buildroot}%{gem_instdir}/extra/dynflow-debug.sh
+ln -s %{gem_instdir}/extra/dynflow-debug.sh %{buildroot}%{foreman_dir}/script/foreman-debug.d/60-dynflow_debug
```
Comment (domcleal) [inline rubygem-foreman-tasks/rubygem-foreman-tasks.spec]: This full foreman-debug.d dir should be created with `%{__mkdir_p}` first, it's likely causing the build failure.

File: rubygem-foreman-tasks/rubygem-foreman-tasks.spec [hunk 7]
```diff
@@ -105,15 +105,20 @@ mv %{buildroot}/%{gem_instdir}/config/%{gem_name}.yaml.example \
...
 %else
-install -Dp -m0644 %{buildroot}%{gem_instdir}/%{confdir}/%{service_name}.sysconfig %{buildroot}%{sysconfig_dir}/%{service_name}
 install -Dp -m0644 %{buildroot}%{gem_instdir}/%{confdir}/%{service_name}.service %{buildroot}%{_unitdir}/%{service_name}.service
 %endif
 mkdir -p %{buildroot}%{bin_dir}
 ln -sv %{gem_instdir}/bin/%{service_name} %{buildroot}%{bin_dir}/%{service_name}
 
+#link dynflow-debug.sh to be called from foreman-debug
+chmod +x %{buildroot}%{gem_instdir}/extra/dynflow-debug.sh
+%{__mkdir_p} %{buildroot}%{foreman_dir}/script
```
Comment (domcleal) [inline rubygem-foreman-tasks/rubygem-foreman-tasks.spec]: This needs to create the last subdir, `foreman-debug.d`.

**Comment by iNecas** [comment]
Right, I hope I got it right now

**Comment by iNecas** [comment]
I wonder if this time it was something intermittent http://koji.katello.org/koji/getfile?taskID=566804&name=root.log

**Comment by iNecas** [comment]
[test]

**Comment by iNecas** [comment]
seems the el6 is failing due to some independent issue

**Comment by iNecas** [comment]
I see it's known issue, sry for spam :)

**Comment by domcleal** [comment]
Yes, it's been left in a mess, sorry.

**Comment by domcleal** [comment]
Merged and building, thanks @iNecas.

**Pull request merged**